recipe/patches/0001-Locate-zoneinfo-based-on-install-prefix.patch

```diff
@@ -0,0 +1,259 @@
+-#  else  // __APPLE__
+-#      if TARGET_OS_IPHONE
+-#          if TARGET_OS_SIMULATOR
+-    return "/usr/share/zoneinfo";
+-#          else
+-    return "/var/db/timezone/zoneinfo";
+-#          endif
+-#      else
+-    CONSTDATA auto timezone = "/etc/localtime";
+-    if (!(lstat(timezone, &sb) == 0 && S_ISLNK(sb.st_mode) && sb.st_size > 0))
+-        throw runtime_error("discover_tz_dir failed\n");
+-    string result;
+-    char rp[PATH_MAX+1] = {};
+-    if (readlink(timezone, rp, sizeof(rp)-1) > 0)
+-        result = string(rp);
+-    else
+-        throw system_error(errno, system_category(), "readlink() failed");
+-    auto i = result.find("zoneinfo");
+-    if (i == string::npos)
+-        throw runtime_error("discover_tz_dir failed to find zoneinfo\n");
+-    i = result.find('/', i);
+-    if (i == string::npos)
+-        throw runtime_error("discover_tz_dir failed to find '/'\n");
+-    return result.substr(0, i);
+-#      endif
+-#  endif  // __APPLE__
++    // get path to library we're in, to determine our location relative to $PREFIX;
++    // with help from the MIT-licensed https://github.com/gpakosz/whereami
++    void* addr =
++#ifdef _WIN32
++    _ReturnAddress()
++#else
++    __builtin_extract_return_addr(__builtin_return_address(0))
++#endif
++    ;
++    char* this_lib;
++    int i;
++    static std::string tz_dir;
++    if (!tz_dir.empty()) {
++      return tz_dir;
++    }
++#ifdef _WIN32
++    char buffer[MAX_PATH];
++    HMODULE hm = NULL;
++    // non-zero return means success
++    if (GetModuleHandleEx(GET_MODULE_HANDLE_EX_FLAG_FROM_ADDRESS |
++                          GET_MODULE_HANDLE_EX_FLAG_UNCHANGED_REFCOUNT,
++                          (LPCSTR) addr, &hm)) {
++      // returns length of string (not counting null byte), see
++      // https://learn.microsoft.com/en-us/windows/win32/api/libloaderapi/nf-libloaderapi-getmodulefilenamew#return-value
++      DWORD total_length = GetModuleFileNameA(hm, buffer, sizeof(buffer));
++      if (total_length) {
++        this_lib = (char*)malloc(total_length + 1);
++        memcpy(this_lib, buffer, total_length);
++#else
++    char buffer[PATH_MAX];
++    Dl_info info;
++
++    if (dladdr(addr, &info)) {
++      char* resolved = realpath(info.dli_fname, buffer);
++      if (resolved) {
++        int total_length = (int)strlen(resolved);
++        this_lib = (char*)malloc(total_length + 1);
++        memcpy(this_lib, resolved, total_length);
++#endif
++
++        for (i = (int)total_length - 1; i >= 0; --i) {
++          if (this_lib[i] == folder_delimiter) {
++            // set to null byte so the string ends before the basename
++            this_lib[i] = '\0';
++            break;
++          }
++        }
++        std::string tmp{this_lib};
++        tmp += folder_delimiter;
++        tmp += "..";
++        tmp += folder_delimiter;
++#ifdef _WIN32
++        tmp += "..";
++        tmp += folder_delimiter;
++#endif
```

Comment:
Is tzdata not located in the library prefix on windows?

Reply:
doesn't seem so - its `%PREFIX%/share/zoneinfo`.

Reply:
you can see this here: https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=1006168&view=logs&j=a70f640f-cc53-5cd3-6cdc-236a1aa90802&t=6119ccbe-9301-594f-7c27-f792b80a7fcc

<img width="707" alt="image" src="https://github.com/user-attachments/assets/82208b4a-5246-4e7a-9c4e-828e9d5259fc">

Reply:
`tzdata` is `noarch: generic`, it is built on `linux_64`, and [its content is installed under `share`](https://github.com/conda-forge/tzdata-feedstock/blob/main/recipe/meta.yaml#L32-L36).